lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -266,6 +273,8 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
         cmd += " --host=%s --port=%i" % (shlex_quote(host), port)
     if all_databases:
         cmd += " --all-databases"
+    elif only_tables and len(db_name) == 1:
```

Comment:
```suggestion
    elif only_tables:
```
probably you don't need this check because you check it in ```main```